From 82c3cddea28401cd634b319dc55bbfae9e8e9014 Mon Sep 17 00:00:00 2001
Message-ID: <82c3cddea28401cd634b319dc55bbfae9e8e9014.1720090727.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1720090727.git.sharlatanus@gmail.com>
References: <cover.1720090727.git.sharlatanus@gmail.com>
From: Rodion Goritskov <rodion.goritskov@gmail.com>
Date: Wed, 3 Jul 2024 23:45:21 +0100
Subject: [PATCH v2 6/9] gnu: Add go-github-com-go-webauthn-x.

* gnu/packages/golang-web.scm (go-github-com-go-webauthn-x): New variable.

Signed-off-by: Sharlatan Hellseher <sharlatanus@gmail.com>
Change-Id: Ia7ebf7cfc9c9e588e9ee968603a5c6ea5237d75d
---
 gnu/packages/golang-web.scm | 36 ++++++++++++++++++++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/gnu/packages/golang-web.scm b/gnu/packages/golang-web.scm
index e64f20801c..bd7d88be6d 100644
--- a/gnu/packages/golang-web.scm
+++ b/gnu/packages/golang-web.scm
@@ -2848,6 +2848,42 @@ (define-public go-github-com-valyala-fasthttp
 replacement for native @code{net/http} module.")
     (license license:expat)))
 
+(define-public go-github-com-go-webauthn-x
+  (package
+    (name "go-github-com-go-webauthn-x")
+    (version "0.1.11")
+    (source
+     (origin
+       (method git-fetch)
+       (uri (git-reference
+             (url "https://github.com/go-webauthn/x")
+             (commit (string-append "v" version))))
+       (file-name (git-file-name name version))
+       (sha256
+        (base32 "020pc8q218q8217c9i346vbing26qpnkidhpjvm5501wq6qm0zak"))))
+    (build-system go-build-system)
+    (arguments
+     (list
+      #:tests? #f ; no tests
+      #:import-path "github.com/go-webauthn/x"
+      #:phases
+      #~(modify-phases %standard-phases
+          ;; Source only package.
+          (delete 'build))))
+    (propagated-inputs
+     (list go-golang-org-x-crypto))
+    (home-page "https://github.com/go-webauthn/x")
+    (synopsis "Low level packages for WebAuthn")
+    (description
+     "This package implements a low level functionality for
+@url{https://github.com/go-webauthn/webauthn,WebAuthn} library.  It was forked
+from CloudFlare's github.com/cloudflare/cfssl/revoke.")
+    (license (list
+              ;; For the CloudFlare's part: revoke/LICENSE.
+              license:bsd-2
+              ;; For the WebAuthn's fork: LICENSE.
+              license:bsd-3))))
+
 (define-public go-github-com-whyrusleeping-json-filter
   (let ((commit "ff25329a9528f01c5175414f16cc0a6a162a5b8b")
         (revision "0"))
-- 
2.41.0

